Update of /cvsroot/xine/xine-lib/src/libxinevdec
In directory sc8-pr-cvs1.sourceforge.net:/tmp/cvs-serv12895

Modified Files:
	image.c 
Log Message:
fix a huge memleak (thanks to Jerome for reporting)


Index: image.c
===================================================================
RCS file: /cvsroot/xine/xine-lib/src/libxinevdec/image.c,v
retrieving revision 1.15
retrieving revision 1.16
diff -u -r1.15 -r1.16
--- src/libxinevdec/image.c	11 Jul 2004 21:38:05 -0000	1.15
+++ src/libxinevdec/image.c	3 Dec 2004 15:46:42 -0000	1.16
@@ -263,7 +263,10 @@
 
 	*out = le2me_16(*out);
       }
+      
+      free(this->rows[row]);
     }
+    free(this->rows);
     _x_stream_info_set(this->stream, XINE_STREAM_INFO_FRAME_DURATION, img->duration);
     img->draw(img, this->stream);
     img->free(img);

